### ISSUE TITLE
Drop support for Python 3.8

### ISSUE THREAD

**Issue by brendan-m-murphy** [openghg/openghg#817]
### What is your issue?

The last version of xarray that supports Python 3.8 was released in January 2023 (version 2023.1.0). In order to unpin pandas (and use versions >= 2.0) we need a slightly more recent version of xarray (>= version 2023.3).

The reason for this is that `xr.Dataset.resample` passes `base` as a kwarg to a pandas `Grouper` object, but this object doesn't support `base` as a kwarg in versions >= 2.0. This was fixed later by the xarray team to raise a `FutureWarning` instead of passing `base = None` to pandas (which causes a `TypeError`).

In short, if we want to unpin pandas and keep using xarray's `resample` method, then we need to drop support for Python 3.8

This issue was brought up in the course of addressing issue #618.
